### ISSUE TITLE
Have problem with sl an

### ISSUE THREAD

**Issue by SamHonarv** [ccxt/ccxt#18456]
### ؜

import ccxt

api_key = 'xxxxxxxxxxxxx'
secret = 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'

exchange = ccxt.mexc3({
    'apiKey': api_key,
    'secret': secret,
    'enableRateLimit': True,
    'options': {'defaultType': 'swap'}
})

positions = exchange.fetch_positions()

if len(positions) == 0:
    print("You don't have any positions")
else:
    for position in positions:
        print(position)
        if 'entryPrice' in position:
            symbol = position['symbol']
            entry_price = float(position['entryPrice'])
            side = position['side']
            tp_price = None
            sl_price = None

            if side == 'long':
                tp_price = entry_price * 1.05  #  Take Profit
                sl_price = entry_price * 0.95  # Stop Loss
            elif side == 'short':
                tp_price = entry_price * 0.95  # Take Profit
                sl_price = entry_price * 1.05  #  Stop Loss

            if tp_price is not None and sl_price is not None:
                #  Take Profit
                tp_order_params = {
                    'symbol': symbol,
                    'type': 'limit',
                    'side': side,
                    'amount': position['collateral'],
                    'price': tp_price,
                    'params': {
                        'takeProfitPrice': tp_price
                    }
                }
                tp_order = exchange.create_order(**tp_order_params)

                #  Stop Loss
                sl_order_params = {
                    'symbol': symbol,
                    'type': 'limit',
                    'side': side,
                    'amount': position['collateral'],
                    'price': sl_price,
                    'params': {
                        'stopLossPrice': sl_price
                    }
                }
                sl_order = exchange.create_order(**sl_order_params)

                print("Take Profit order created:", tp_order)
                print("Stop Loss order created:", sl_order)

I wrote this code, it easily checks everything and gives the position information, but it can't set the SL and TP, I changed the code in every way, I used different methods, but the code doesn't work, please help me.

**Comment by samgermain** [comment]
The params `stopLossPrice` and `takeProfitPrice` are both meant to be passed via createOrder when you intially create a position. You can also use the param `triggerPrice` to create a separate stop order which is not explicitly attached to your position

[You can read about the differences between these parameters on this post ](https://github.com/ccxt/ccxt/issues/13822)

**Comment by SamHonarv** [comment]
Yes, I considered all of these things, but the MEXC exchange does not allow SL and TP to be set under any circumstances, I wrote the code for the Binance exchange many times, but there was no problem, but I have been trying for 5 days now, but it doesn't work.

**Comment by ttodua** [comment]
@SamHonarv I've submited a PR that adds unification of SL & TP for mexc, so you will be able to use it once it's merged.

**Comment by ttodua** [comment]
by the way, are you able to place an order through API on mexc? if so, how? as they have disabled it currently (your issue was opened 3 weeks ago, so maybe by that time it was enabled)

**Comment by carlotestor** [comment]
The main reason that code can't work: `create_order(**params)` with a nested `'params'` key isn't the right call shape — `params` must be the 6th argument, not a dict key, and `amount` must be the position size in contracts, not `position['collateral']`. As written, MEXC receives a plain limit order with no trigger at all.

Correct shape for a reduce-only SL/TP on an open position:

```python
symbol = position['symbol']
amount = position['contracts']
close_side = 'sell' if position['side'] == 'long' else 'buy'

# take profit
exchange.create_order(symbol, 'market', close_side, amount, None,
                      {'takeProfitPrice': tp_price, 'reduceOnly': True})

# stop loss
exchange.create_order(symbol, 'market', close_side, amount, None,
                      {'stopLossPrice': sl_price, 'reduceOnly': True})
```

Two important caveats for MEXC specifically:
- Use `ccxt.mexc`, not `ccxt.mexc3` (that alias is long gone).
- **MEXC has had contract (futures) API trading disabled for most accounts since 2022** — order placement on swap markets returns an error regardless of the client. That's the wall you hit for 5 days, and no CCXT-side change fixes it. Check `exchange.features['swap']['linear']['createOrder']` to see what's unified, but account-level API access is the gating factor here.

If you need working futures SL/TP today, Binance/Bybit/Bitget all support the exact snippet above.
